apps/dav/lib/CardDAV/VCFMultiExportPlugin.php

```diff
@@ -0,0 +1,190 @@
+<?php
+declare (strict_types = 1);
+/**
+ * @copyright Copyright (c) 2018 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\DAV\CardDAV;
+
+use OCA\DAV\CardDAV\AddressBook;
+use Sabre\DAV;
+use Sabre\DAV\Server;
+use Sabre\HTTP\RequestInterface;
+use Sabre\HTTP\ResponseInterface;
+use Sabre\VObject;
+
+class VCFMultiExportPlugin extends DAV\ServerPlugin {
+
+	/** @var Server */
+	protected $server;
+
+	/**
+	 * Initializes the plugin and registers event handlers
+	 *
+	 * @param Server $server
+	 * @return void
+	 */
+	public function initialize(Server $server) {
+		$this->server = $server;
+		$this->server->on('method:GET', [$this, 'httpGet'], 90);
+	}
+
+	/**
+	 * Intercepts GET requests
+	 *
+	 * @param RequestInterface $request
+	 * @param ResponseInterface $response
+	 * @return bool
+	 */
+	public function httpGet(RequestInterface $request, ResponseInterface $response) {
+
+		$queryParams = $request->getQueryParameters();
+
+		// check for post data validity
+		if (!array_key_exists('vcards', $queryParams)
+		 || !is_array($queryParams['vcards'])) {
+			return true;
+		}
+
+		// user addressbooks home
+		$path = $request->getPath();
+		$paths = $queryParams['vcards'];
+
+		// extract unique addressbooks from all the paths
+		$addressbookIDs = array_unique(array_map(function($path) {
+			return explode('/', $path)[0];
+		}, $paths));
+
+		// valid addressbooks and the requested vcards
+		$addressbooks = [];
+		foreach ($addressbookIDs as $addressbook) {
+			$node = $this->server->tree->getNodeForPath("$path/$addressbook/");
+
+			// Checking ACL, if available.
+			if ($aclPlugin = $this->server->getPlugin('acl')) {
+				$aclPlugin->checkPrivileges("$path/$addressbook/", '{DAV:}read');
+			}
+
+			// checking that the path is a valid addressbook
+			if ($node instanceof AddressBook) {
+				$addressbooks = array_merge($addressbooks, [$addressbook => $vcards]);
```

Comment:
```suggestion
				$addressbooks[$addressbook] = $vcards;
```

would do the exact same I guess

Reply:
Actually I can drop this I think now. I don't use $addressbooks anymore! :+1: